adobe/spectrum-web-components
# fix(menu): add support for submenu interactions

## Description
Add support for submenu content in `<sp-menu-item>` elements.

### To dos:
- [x] review `<sp-menu>` width application in submenus as it relates to breaking in the middle of words
- [x] maintain "active" state on the anchor element of open submenus
- [x] ensure ALL menus are closed when clicking away after opening multiple submenus
- [x] prevent multiple sibling submenus being open at the same time
- [x] ensure submenus reopen after clicking on them the first time
- [x] expand stories
  - [x] explain eventing
  - [x] as "context menu"
- [x] review file size/performance side effects
  - the perf diff seems reasonable here
  - the file size diff seems a little larger than I expected, there may be room to further optimize this, but it doesn't feel like it should be a block at this stage

## Related issue(s)
- fixes #146

## Motivation and context
Deliver more expansive menu content accessibly.

## How has this been tested?

    1. Go [here](https://sub-menu--spectrum-web-components.netlify.app/storybook/?path=/story/menu-submenu--submenu)
    2. Expect the layout of the second submenu to be a little bit wonky to start, it's opened by default for the VRTs which account for the timing needed to make this work correctly, [see screen shot](https://c166d9d48e0a89bf135470b620d0cec6--spectrum-web-components.netlify.app/review/#SubmenuStories/submenu.png).
    3. Click away from the menus
    4. See that the ALL close
    5. Click the hamburger button
    6. Mouse in and out of the various menus and make a selection of `Brooklyn > Ft. Greene > S. Elliot Pl`
    7. See that the selection is represented in the page.
    1. Go [here](https://sub-menu--spectrum-web-components.netlify.app/storybook/?path=/story/menu-submenu--submenu)
    2. Expect the layout of the second submenu to be a little bit wonky to start, it's opened by default for the VRTs which account for the timing needed to make this work correctly, [see screen shot](https://c166d9d48e0a89bf135470b620d0cec6--spectrum-web-components.netlify.app/review/#SubmenuStories/submenu.png).
    3. Click away from the menus
    4. See that the ALL close
    5. Tab to the hamburger menu
    6. Press `Space`
    7. See the menu open
    8. Press `ArrowDown`
    9. See "Brooklyn" focused
    8. Press `ArrowDown`
    9. See "Manhattan" focused
    10. Press `ArrowRight`
    11. See the submenu open and "Union Square" focused
    12. Press `ArrowRight`
    13. See the submenu open and "14th St" focused
    14. Press `Enter`
    15. See ALL the menus close
    16. See the selection represented in the page
    1. Go [here](https://sub-menu--spectrum-web-components.netlify.app/storybook/?path=/story/menu-submenu--context-menu)
    2. Right click the page
    3. Use various selection methods (pointer or keyboard) to choose `Select Layer > Rectangle`
    4. See the selection represented in the page
    1. Go [here](https://sub-menu--spectrum-web-components.netlify.app/components/menu-item/#submenu)
    2. Interact with the submenu demo
    3. See that is opens/closes via expected interactions (pointer and keyboard)

## Types of changes

## Checklist
